bundles/binding/org.openhab.binding.anel/src/main/java/org/openhab/binding/anel/internal/AnelDataParser.java

```diff
@@ -98,8 +101,80 @@
 
         final Map<AnelCommandType, State> result = new LinkedHashMap<AnelCommandType, State>();
 
-        // check for switch changes, update cached state, and prepare command if
-        // needed
+        // maybe the device's name changed?!
+        checkForNameChange(arr, state, result);
+
+        if (arr.length >= 15) {
+            checkForStateChanges(arr, state, result);
+        }
+
+        // IO is occupies fields 16-23
+        if (arr.length >= 24) {
+            checkForIOChanges(arr, state, result);
+        }
+
+        // IO and temperature is only available if array has at least length 25
+        if (arr.length >= 25) {
+            checkForTemperatureChange(arr, state, result);
+        }
+
+        /*
+         * subsequent fields may be firmware, power measurement (is that a variable count of fields?), sensor data,
+         * and password encryption information.
+         * to read the sensor values, we have to search for the pattern 's:<temperature>:<humidity>:<brightness>'.
+         * corresponding github issue and Anel protocol documentation:
+         * https://github.com/openhab/openhab1-addons/issues/5338
+         * https://anel-elektronik.de/forum_neu/viewtopic.php?f=16&t=207
+         */
+        final int sensorDataIndex = findSensorDataIndex(arr, 26);
```

Comment:
Why not final?

Reply:
My question would be why final? It seems to serve no purpose.

Reply:
Yes it does. Final ensures that a variable can not be changed any more. Hence, its purpose remains the same and it cannot be (ab)used for different purposes. From my experience this typically results in cleaner code.

Reply:
I don't buy any of that. However, I've since done some googling and I see other people making arguments in favor of it. 
So I'm ok with keeping these uses of `final`.